sigar_windows.go

```diff
@@ -542,11 +542,103 @@ func (self *NetUdpV6ConnList) Get() error {
 }
 
 func (self *NetProtoV4Stats) Get() error {
-	return notImplemented()
+	protoV4Queries := []string{
+		`\TCPv4\Connections Active`,
+		`\TCPv4\Connections Passive`,
+		`\TCPv4\Connection Failures`,
+		`\TCPv4\Connections Reset`,
+		`\TCPv4\Connections Established`,
+
+		`\UDPv4\Datagrams Received Errors`,
+
+		`\IPv4\Datagrams Received Header Errors`,
+		`\IPv4\Datagrams Received Address Errors`,
+		`\IPv4\Datagrams Received Discarded`,
+		`\IPv4\Datagrams Received Unknown Protocol`,
+		`\IPv4\Datagrams Outbound Discarded`,
+		`\IPv4\Datagrams Outbound No Route`,
+
+		`\ICMP\Messages Received Errors`,
+		`\ICMP\Received Dest. Unreachable`,
+		`\ICMP\Messages Outbound Errors`,
+		`\ICMP\Sent Destination Unreachable`,
+	}
+
+	results, err := runRawPdhQueries(protoV4Queries)
+	if err != nil {
+		return err
+	}
+	self.TCP.ActiveOpens, results = uint64(results[0]), results[1:]
```

Comment:
This idiom is a pretty interesting way to iterate through the array.  I like it more than the C-styled `results[i++]`.  Array slices are just views into the array, not copies, right?

Is it possible for `results` not to have enough values?  In other words, if a call to `results[1:]` would prematurely return an empty array, would the subsequent `results[0]` call then cause an error?